src/logic/wallets/transactionDataCheck.ts

```diff
@@ -1,18 +1,19 @@
+import { loadFromStorage, saveToStorage } from 'src/utils/storage'
+
+const LEDGER_TRANSACTION_DATA = 'TRANSACTION_DATA'
+const WALLET = 'Ledger'
 function transactionDataCheck(): any {
-  let completed = false
-  return (stateAndHelpers) => {
+  return async (stateAndHelpers) => {
     const { wallet } = stateAndHelpers
-
-    if (wallet && wallet.name === 'Ledger' && !completed) {
+    const isTransactionDataEnabled = await loadFromStorage(LEDGER_TRANSACTION_DATA)
```

Comment:
for loadFromStorage you can use a type argument to express what type you're expecting, e.g `loadFromStorage<boolean>(LEDGER_TRANSACTION_DATA)`